### PR TITLE
OCC-4-alt: Add UI testing with a basic test

### LINKED ISSUE #110: Add UI testing with a basic test
The goal is to write UI tests during the development of new features and bug fixes. For that, we need to add https://github.com/Lombiq/UI-Testing-Toolbox. Add a basic initial UI test: https://github.com/Lombiq/UI-Testing-Toolbox/blob/dev/Lombiq.Tests.UI.Samples/Tests/BasicOrchardFeaturesTests.cs and make sure it's being tested during the build in the CI when creating or updating a pull request.

### PR DESCRIPTION
**Pull request by Psichorex**
OCC-4
Fixes #110

### CODE REVIEW

File: test/OrchardCore.Commerce.Tests.UI/Config.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿namespace OrchardCore.Commerce.Tests.UI;
+
+public static class Config
+{
+    public static string GetAbsoluteApplicationAssemblyPath()
+    {
+        // The test assembly can be in a folder below the src and test folders (those should be in the repo root).
+        var baseDirectory = File.Exists("SnowMountain.Web.dll")
```
Comment (DAud-IcI) [inline test/OrchardCore.Commerce.Tests.UI/Config.cs:8]: Wrong project.

File: test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using Lombiq.Tests.UI.Extensions;
+using Lombiq.Tests.UI.Pages;
+using Lombiq.Tests.UI.Services;
+using OpenQA.Selenium;
+
+namespace OrchardCore.Commerce.Tests.UI.Helpers;
+
+public static class SetupHelpers
+{
+    public const string RecipeId = "Blog";
```
Comment (DAud-IcI) [inline test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs:10]: Please add a setup recipe that calls Blog as well as the Product and Order recipes and use it here. This is to prove the OCC features don't interfere with the basic functionality.

**Review by DAud-IcI — CHANGES_REQUESTED** [review]

File: test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs (hunk 2 above)
Comment (Psichorex) [reply test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs:10]: Btw I wanted to use Sofware as a Service but for some reason it produced an error while the Blog is working fine.

File: test/OrchardCore.Commerce.Tests.UI/Config.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            ? AppContext.BaseDirectory
+            : Path.Combine(
+                AppContext.BaseDirectory.Split(new[] { "src", "test" }, StringSplitOptions.RemoveEmptyEntries)[0],
+                "src",
+                "SampleWebApp",
+                "bin",
+                "Debug",
+                "net6.0");
+
+        return Path.Combine(baseDirectory, "SnowMountain.Web.dll");
```
Comment (DAud-IcI) [inline test/OrchardCore.Commerce.Tests.UI/Config.cs:18]: Here too.

File: test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs (hunk 2 above)
Comment (DAud-IcI) [reply test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs:10]: The recipe ID here remains the same so the recipe you created isn't actually used.

File: OrchardCore.Commerce/Recipes/OrchardCore.Commerce.Setup.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  "displayName": "OrchardCore Commerce",
+  "description": "Main setup recipe for the OrchardCore Commerce platform.",
+  "author": "OrchardCoreCommerce",
+  "website": "https://www.orchardcorecommerce.com",
+  "version": "1.0",
+  "issetuprecipe": true,
+  "categories": ["orchardcorecommerce"],
+  "tags": [ "setup" ],
+
+  "steps": [
```
Comment (DAud-IcI) [inline OrchardCore.Commerce/Recipes/OrchardCore.Commerce.Setup.json:12]: Like I said, execute the OCC-specific recipes:
- https://github.com/OrchardCMS/OrchardCore.Commerce/blob/main/OrchardCore.Commerce/Recipes/Order.recipe.json
- https://github.com/OrchardCMS/OrchardCore.Commerce/blob/main/OrchardCore.Commerce/Recipes/Product.recipe.json

File: OrchardCore.Commerce/Recipes/OrchardCore.Commerce.Setup.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿{
```
Comment (DAud-IcI) [inline OrchardCore.Commerce/Recipes/OrchardCore.Commerce.Setup.json:1]: The file name should end with `.recipe.json`.

**Review by DAud-IcI — CHANGES_REQUESTED** [review]

File: test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs (hunk 2 above)
Comment (Psichorex) [reply test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs]: Adding the recipe to the recipeId breaks the tests. This is my first time setting up this whole UI test system and recipes.
Could you please help me spot the problem? I observed FailureDumps and it doesn't get past the setup site stage.

Also it couldn't set the recipe to blog.

Comment (Psichorex) [reply test/OrchardCore.Commerce.Tests.UI/Helpers/SetupHelpers.cs]: The is pushed also

**Pull request merged**